BeagleBone_black/src/state_machine/main.hpp

```diff
@@ -44,15 +44,12 @@ class Main: public Thread {
  private:
   HypedMachine hypedMachine;
   data::Data& data_;
-  /**
-   * @brief      Checks all the system for critical failures. Returns false if all checked have passed.
-   */
-  bool hasCriticalFailure();
-  /**
-   * @brief      Computes the maxDistance and returns true if it has been reached.
-   */
-  bool hasReachedMaxDistance();
+  data::Communications comms_data;
+  data::Navigation nav_data;
+  data::StateMachine sm_data;
+  void checkNavigation();
```

Comment:
I advice to separate functions from data. Have data declared at the very end of the class declaration.